transport/http/src/main/java/org/kaazing/gateway/transport/http/DefaultHttpSession.java

```diff
@@ -615,5 +626,22 @@ public boolean isHttpxeSpecCompliant() {
         return httpxeSpecCompliant;
     }
 
+        void resetParent(IoSessionEx newParent){
+        setParent(newParent);
+    }
+
+    int getAndDecrementRedirectsAllowed() {
```

Comment:
Yeah, without it the parent is wrong and write events are going down the wrong downstream